wapitiCore/attack/mod_http_post.py

```diff
@@ -0,0 +1,36 @@
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import MEDIUM_LEVEL, _
+from wapitiCore.definitions.http_post import NAME
+
+
+# This module check the security of transported credentials of login forms
+class mod_http_post(Attack):
+    """Check if credentials are transported on an encrypted channel."""
+    name = "http_post"
+
+    async def must_attack(self, request: Request):
+        # We leverage the fact that the crawler will fill password entries with a known placeholder
+        if "https://" in request.url:
+            return False
+
+        return True
+
+    async def attack(self, request: Request):
+        
+        if "Letm3in_" not in request.encoded_data + request.encoded_params:
+            return
+        self.finished = True
```

Comment:
I think you can remove this line

Reply:
Also as an improvement we can keep track of files we marked as vulnerable (using `request.path`) to prevent duplicates (you can try on http://www.tvsoop.com/ for example which will generate lot of duplicates)